lib/smart_answer_flows/uk-benefits-abroad.rb

```diff
@@ -601,14 +592,151 @@ def define
+              end
+            end
+          when "after_jan_2021"
+            if calculator.benefit == "jsa"
+              outcome :jsa_not_entitled_outcome
+            elsif calculator.benefit == "winter_fuel_payment"
+              outcome :wfp_not_eligible_outcome
+            elsif calculator.benefit == "esa"
+              if calculator.going_abroad
+                outcome :esa_going_abroad_other_outcome
+              else
+                outcome :esa_already_abroad_other_outcome
+              end
+            elsif calculator.benefit == "disability_benefits"
+              if calculator.going_abroad
+                outcome :db_going_abroad_other_outcome
+              else
+                outcome :db_already_abroad_other_outcome
+              end
+            end
+          when "no"
+            if calculator.benefit == "jsa"
+              outcome :jsa_not_entitled_outcome
+            elsif calculator.benefit == "winter_fuel_payment"
+              outcome :wfp_not_eligible_outcome
+            elsif calculator.benefit == "esa"
+              if calculator.going_abroad
+                outcome :esa_going_abroad_other_outcome
+              else
+                outcome :esa_already_abroad_other_outcome
+              end
+            elsif calculator.benefit == "disability_benefits"
+              if calculator.going_abroad
+                outcome :db_going_abroad_other_outcome
+              else
+                outcome :db_already_abroad_other_outcome
+              end
+            end
+          end
+        end
+      end
+
+      radio :is_british_or_irish? do
+        option :yes
+        option :no
+
+        next_node do |response|
+          case response
+          when "yes"
+            if calculator.benefit == "jsa"
+              outcome :jsa_ireland_outcome
+            elsif calculator.benefit == "winter_fuel_payment"
+              outcome :wfp_ireland_outcome
+            elsif calculator.benefit == "esa"
+              if calculator.going_abroad
+                outcome :esa_going_abroad_eea_outcome
+              else
+                outcome :esa_already_abroad_eea_outcome
+              end
+            elsif calculator.benefit == "disability_benefits"
+              outcome :db_going_abroad_ireland_outcome
+            end
+          when "no"
+            question :worked_in_eea_or_switzerland?
+          end
+        end
+      end
+
+      radio :how_long_abroad? do
+        option :one_year_or_less
+        option :more_than_one_year
+
+        next_node do |response|
+          case response
+          when "one_year_or_less"
+            if calculator.channel_islands?
+              outcome :jsa_channel_islands_outcome
+            else
+              outcome :jsa_social_security_going_abroad_outcome
+            end
+          when "more_than_one_year"
+            outcome :jsa_social_security_going_abroad_outcome
+          end
+        end
```

Comment:
With only two cases, and with a conditional that otherwise gives a default outcome, I think this could be simplified:

```ruby
next_node do |response|
  if response == "one_year_or_less" && calculator.channel_islands?
    outcome :jsa_channel_islands_outcome 
  else
    outcome :jsa_social_security_going_abroad_outcome
  end
end
```